src/SimpleEventStore/SimpleEventStore.AzureDocumentDb/Resources/appendToStream.js

```diff
@@ -1,49 +1,63 @@
-function appendToStream(documents) {
+function appendToStream(events) {
     var context = getContext();
     var collection = context.getCollection();
-    var collectionLink = collection.getSelfLink();
-    var streamId = documents[0].streamId;
-    var expectedEventNumber = documents[0].eventNumber - 1;
+    var streamId = events[0].streamId;
+    var firstEventNumber = events[0].eventNumber;
 
-    var concurrencyQuery = {
-        query: "SELECT TOP 1 c.eventNumber FROM Commits c WHERE c.streamId = @streamId ORDER BY c.eventNumber DESC",
-        parameters: [{ name: "@streamId", value: streamId }]
-    };
-
-    var accepted = collection.queryDocuments(collectionLink, concurrencyQuery, {}, onConcurrencyQueryCompleted);
+    if (firstEventNumber == 1) {
+        createEvents();
+    }
+    else {
+        createEventsOnlyIfPreviousEventExists();
+    }
 
-    if (!accepted) throw new Error("Concurrency query not accepted.");
+    function createEventsOnlyIfPreviousEventExists() {
+        var previousEventNumber = firstEventNumber - 1;
+        var previousDocId = streamId + ":" + previousEventNumber;
+        var previousDocLink = collection.getAltLink() + '/docs/' + previousDocId;
 
-    function onConcurrencyQueryCompleted(err, doc, options) {
-        if (err) throw err;
+        var isAccepted = collection.readDocument(previousDocLink, {}, onPreviousEventCompleted);
+        if (!isAccepted) {
+            throw new Error(500, "Existing event read not accepted for creation.");
+        }
 
-        var actualEventNumber = doc.length > 0 ? doc[0].eventNumber : 0;
+        function onPreviousEventCompleted(err, doc, options) {
```

Comment:
Nit: not sure about the naming of this function. It's a handler rather than an event so I think it'd be clearer if named something like `handlePreviousEventResponse`.

Reply:
I'm going to stick with the `on...` convention - this is what's used in most of the samples I've looked at.